File: solutions/python/02.py
def createFrame(names):
    orderedNames = sorted(names, key=lambda name: len(name))
    longestName = orderedNames[-1]
    drawLength = len(longestName) + 1
    frameBorder = "*" * (2 + drawLength + 1)
    frame = ""
    frame += frameBorder + "\n"
    for name in names:
        frame += f"* {name.ljust(drawLength)}*\n"
    frame += frameBorder
    return frame


if __name__ == "__main__":
    frame1 = createFrame(["midu", "madeval", "educalvolpz"])
    print(frame1)
    # Expected result:
    """
    ***************
    * midu        *
    * madeval     *
    * educalvolpz *
    ***************
    """

    frame2 = createFrame(["midu"])
    print(frame2)
    # Expected result:
    """
    ********
    * midu *
    ********
    """

    frame3 = createFrame(["a", "bb", "ccc"])
    print(frame3)
    # Expected result:
    """
    *******
    * a   *
    * bb  *
    * ccc *
    *******
    """

    frame4 = createFrame(["a", "bb", "ccc", "dddd"])
    print(frame4)

"""
  Score: ★★★★★
"""

File: solutions/js/02.js
function createFrame(names) {
  const orderedNames = [...names].sort((a, b) => a.length - b.length);
  const longestName = orderedNames[orderedNames.length - 1];
  const drawLength = longestName.length + 1;
  const frameBorder = "*".repeat(2 + drawLength + 1);
  let frame = "";
  frame += frameBorder + "\n";
  names.forEach((name) => (frame += `* ${name.padEnd(drawLength)}*\n`));
  frame += frameBorder;
  return frame;
}

const frame1 = createFrame(["midu", "madeval", "educalvolpz"]);
console.log(frame1);
// Expected result:
/*
 ***************
 * midu        *
 * madeval     *
 * educalvolpz *
 ***************
 */

const frame2 = createFrame(["midu"]);
console.log(frame2);
// Expected result:
/*
 ********
 * midu *
 ********
 */

const frame3 = createFrame(["a", "bb", "ccc"]);
console.log(frame3);
// Expected result:
/*
 *******
 * a   *
 * bb  *
 * ccc *
 *******
 */

const frame4 = createFrame(["a", "bb", "ccc", "dddd"]);
console.log(frame4);

/* 
  Score: ★★★★★
*/
